components/datetime/src/format/datetime.rs

```diff
@@ -99,45 +103,65 @@ where
-    MissingEraSymbol(FormattingEra),
-    /// Missing weekday symbol
-    #[displaydoc("Cannot find symbol for weekday {0:?}")]
-    MissingWeekdaySymbol(IsoWeekday),
-
-    // Invalid input
-    /// Incomplete input
+    /// An input field (such as "hour" or "month" is missing).
+    ///
+    /// This is guaranteed not to happen for `icu::calendar` inputs, but may happen for custom inputs.
+    ///
+    /// The output will contain the string `{X}` instead, where `X` is the symbol for which the input is missing.
     #[displaydoc("Incomplete input, missing value for {0:?}")]
     MissingInputField(&'static str),
-    /// Cyclic year overflow
-    #[displaydoc("Cyclic year overflow, found {value}, maximum {max}")]
-    CyclicYearOverflow {
+    /// The [`MonthCode`] of the input is not valid for this calendar.
+    ///
+    /// This is guaranteed not to happen for `icu::calendar` inputs, but may happen for custom inputs.
+    ///
+    /// The output will contain the raw [`MonthCode`] as a fallback value.
+    #[displaydoc("Invalid month {0:?}")]
+    InvalidMonthCode(MonthCode),
+    /// The [`FormattingEra`] of the input is not valid for this calendar.
+    ///
+    /// This is guaranteed not to happen for `icu::calendar` inputs, but may happen for custom inputs.
+    ///
+    /// The output will contain [`FormattingEra::fallback_name`] as the fallback.
+    #[displaydoc("Invalid era {0:?}")]
+    InvalidEra(FormattingEra),
+    /// The [`YearInfo::cyclic`] of the input is not valid for this calendar.
+    ///
+    /// This is guaranteed not to happen for `icu::calendar` inputs, but may happen for custom inputs.
+    ///
+    /// The output will contain [`YearInfo::extended_year`] as a fallback value.
+    #[displaydoc("Invalid cyclic year {value} (maximum {max})")]
+    InvalidCyclicYear {
         /// Value
         value: usize,
         /// Max
         max: usize,
     },
+
+    /// The [`FixedDecimalFormatter`] has not been loaded.
+    ///
+    /// This *only* happens if the formatter has been created using
+    /// [`TypedDateTimeNames::with_pattern`], the pattern requires decimal
+    /// formatting, and the decimal formatter was not loaded.
+    ///
+    /// The output will contain fallback values using Latin numerals.
+    #[displaydoc("FixedDecimalFormatter not loaded")]
+    FixedDecimalFormatterNotLoaded,
+    /// The localized names for a field have not been loaded.
+    ///
+    /// This *only* happens if the formatter has been created using
+    /// [`TypedDateTimeNames::with_pattern`], and the pattern requires names
+    /// that were not loaded.
+    ///
+    /// The output will contain fallback values using field identifiers (such as `tue` for `IsoWeekday::Tuesday`,
+    /// `M02` for month 2, etc.).
+    #[displaydoc("Names for {0:?} not loaded")]
+    NamesNotLoaded(Field),
     /// Unsupported field
+    ///
+    /// This *only* happens if the formatter has been created using
```

Comment:
is this something we can validate during construction so that this is not an error?

Reply:
#5773